convert2rhel/subscription.py

```diff
@@ -281,19 +281,28 @@ def attach_subscription():
         tool_opts.pool = pool
         loggerinst.info("Attaching provided subscription pool ID to the system ...")
     else:
-        # Let the user choose the subscription appropriate for the conversion
-        loggerinst.info("Manually select subscription appropriate for the conversion")
         subs_list = get_avail_subs()
+
         if len(subs_list) == 0:
             loggerinst.warning("No subscription available for the conversion.")
             return False
 
-        print_avail_subs(subs_list)
-        sub_num = utils.let_user_choose_item(len(subs_list), "subscription")
-        pool = "--pool " + subs_list[sub_num].pool_id
-        tool_opts.pool = pool
-        loggerinst.info("Attaching subscription with pool ID %s to the system ..." % subs_list[sub_num].pool_id)
+        elif len(subs_list) == 1:
+            sub_num = 0
+            loggerinst.info(
+                " %s is the only subscription available, it will automatically be selected for the conversion."
+                % subs_list[0].pool_id
+            )
 
+        else:
+            # Let the user choose the subscription appropriate for the conversion
+            loggerinst.info("Manually select subscription appropriate for the conversion")
+            print_avail_subs(subs_list)
+            sub_num = utils.let_user_choose_item(len(subs_list), "subscription")
+            loggerinst.info("Attaching subscription with pool ID %s to the system ..." % subs_list[sub_num].pool_id)
+
+    pool = "--pool " + subs_list[sub_num].pool_id
+    tool_opts.pool = pool
```

Comment:
Should be indented one more. Also throws an error if you try to convert.
```suggestion
        pool = "--pool " + subs_list[sub_num].pool_id
        tool_opts.pool = pool
```

Since no tests failed either due to the pool being incorrect/not working you should add another test to verify that the `tool_opts.pool` is set correctly if `tool_opts.auto_attach` or `tool_opts.pool` is set (it should throw an error but don't make assertions on this since it's related to the bug and not implementation). In this case it's better to write the test first, see if it fails, then indent to see if the tests pass